### PR TITLE
fix: assertEqual "diffi better for object getters"

### LINKED ISSUE #1767: assertEquals' diff: does not show actual values compared from getters
**Describe the bug**

The diff printed whenever `assertEquals` fails does not include the actual values compared from enumerable properties implemented using [getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get).

**Steps to Reproduce**

1. Create a test file with following contents:

	  ```ts
	  import { assertEquals } from "https://deno.land/std@0.119.0/testing/asserts.ts";
	  
	  Deno.test("example", () => {
	    assertEquals(
	      Object.defineProperty({}, "a", {
	        enumerable: true,
	        get() {
	          return 1;
	        },
	      }),
	      Object.defineProperty({}, "a", {
	        enumerable: true,
	        get() {
	          return 2;
	        },
	      })
	    );
	  });
	  ```
2. Run the test with `deno test`
3. Observe the diff printed and how there is no meaningful indicator of what wasn't equal:
	```console
	      [Diff] Actual / Expected
	  
	  
	      {
	        a: [Getter],
	      }
	```

**Expected behavior**

A diff printed showing the values that were compared and determined to not be equal. e.g.:

```console
    [Diff] Actual / Expected


    {
-     a: 1,
+     a: 2,
    }
```

or if it is important to show that the values were acquired via getters then something like the following might be printed:

```console
    [Diff] Actual / Expected


    {
-     a: <Getter> 1,
+     a: <Getter> 2,
    }
```

**Environment**

- OS: MacOS 12.0.1
- deno version: 1.17.1 (release, x86_64-apple-darwin)
- std version: 0.119.0

**Remarks**

This might be similar to https://github.com/denoland/deno_std/issues/1020 in that it relies on `Deno.inspect` upstream and that if it were to expose the actual values then `assertEquals'`s diff should work but it is unclear to me if `Deno.inspect` should go as far as calling getters or not. A getter could return a completely different value from call-to-call so my instincts tell me that the onus probably lies with the `assert` module to keep track of the values actually compared to provide in a meaningful diff and not done separately as that could cause the diff to show a different value than what was actually used during the assertion.

### PR DESCRIPTION
**Pull request by cuobiezi**
closes #1767

### CODE REVIEW

**Review by kt3k — COMMENTED** [review]
@cuobiezi Thanks for the PR! Left one inline comment

Can you also add a simple test case in `testing/_format_test.ts` which checks the formatting of the example given in #1767

File: testing/_format.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@
  * make tests less flaky
  * @param v Value to be formatted
  */
-export function format(v: unknown): string {
+export function format(v: unknown, getters?: boolean): string {
```
[the remark this answers is not in the document]
Comment (cuobiezi) [reply testing/_format.ts:9]: I'm not sure too , and I don't know whether other format caller can apply getters: true, So I set this opt for least affect. let me check it . Thanks

[the remark this answers is not in the document]
Comment (cuobiezi) [reply testing/_format.ts]: Your are right , updated

**Comment by cuobiezi** [comment]
@kt3k done

**Review by kt3k — APPROVED** [review]
@cuobiezi LGTM. Nice work!

**Pull request merged**